src/main/java/tictactoe/GameSaver.java

```diff
@@ -0,0 +1,99 @@
+package tictactoe;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Arrays;
+
+public class GameSaver {
+
+    private final Connection connection;
+
+    public GameSaver(Connection connection) {
+        this.connection = connection;
+    }
+
+    public void saveGame(String gameName, Game game) {
+        var gameState = getGameState(gameName, game);
+        var squares = getSquares(game);
+        addEntryToDatabase(gameState, squares);
+    }
+
+    private void addEntryToDatabase
+            (ArrayList<String> gameState, String[] squares) {
+        var query = "INSERT INTO saved_games ("
+                + " game_name,"
+                + " current_player_mark,"
+                + " current_player_type,"
+                + " opponent_mark,"
+                + " opponent_type,"
+                + " squares ) VALUES ("
+                + "?, ?, ?, ?, ?, ?)";
+
+        try {
+            PreparedStatement preparedStatement = connection.prepareStatement(query);
+            var squaresArray = connection.createArrayOf("text", squares);
+
+            preparedStatement.setString(1, gameState.get(0));
```

Comment:
There is a lot of repetition going on there, did you think about using loop to rewrite it? And if not about loop, then about wrapping the `.setString` login in a method to make the code more descriptive?

Reply:
Thanks, Justyna. I'll look into this :)